### ISSUE TITLE
ChatGPT API stopped working with error message "That model does not exist"

### ISSUE THREAD

**Issue by SimonBaars** [transitive-bullshit/chatgpt-api#311]
### Verify latest release

- [X] I verified that the issue exists in the latest `chatgpt` release (confirmed on 4.1.3)

### Verify webapp is working

- [X] I verify that the ChatGPT webapp is working properly for this account.

### Environment details

```
[simon@simon ~]$ node --version 
v19.6.0
[simon@simon Music]$ hostnamectl
 Static hostname: simon
       Icon name: computer-laptop
         Chassis: laptop 💻
         Kernel: Linux 5.15.91-4-lts
    Architecture: x86-64
 Hardware Vendor: Dell Inc.
  Hardware Model: XPS 15 7590
Firmware Version: 1.7.0
```

### Describe the Bug

Earlier today, this library still worked, now even the basic example gives the following error:
```
/usr/bin/node ./src/test-api.js
Uncaught ChatGPTError Error: ChatGPT error 404: {
    "error": {
        "message": "That model does not exist",
        "type": "invalid_request_error",
        "param": null,
        "code": null
    }
}
    at <anonymous> (/home/simon/git/Misc-Projects/consequences.app/node_modules/chatgpt/build/index.js:202:29)
    at processTicksAndRejections (internal/process/task_queues:95:5)
    --- await ---
    at processTicksAndRejections (internal/process/task_queues:95:5)
    --- await ---
    at runMainESM (internal/modules/run_main:53:21)
    at executeUserEntryPoint (internal/modules/run_main:79:5)
    at <anonymous> (internal/main/run_main_module:23:47)
Process exited with code 1
```

This runs the following code:
```
import { ChatGPTAPI } from 'chatgpt'

const api = new ChatGPTAPI({
  apiKey: process.env.OPENAI
})

const res = await api.sendMessage('Hello World!')
console.log(res.text)
```

It worked earlier today. Might something have changed with the API?

**Comment by iota-unum** [comment]
yes, sadly openai shut down the model

**Comment by christophebe** [comment]
yes, this is the same issue as https://github.com/transitive-bullshit/chatgpt-api/issues/292

Is it possible to find the new model name somewhere ?

**Comment by transitive-bullshit** [comment]
We're working on a workaround.

In the meantime, you can switch back to using `text-davinci-003` via https://github.com/transitive-bullshit/chatgpt-api/releases/tag/v4.2.0

**Comment by gigantino** [comment]
> We're working on a workaround.
> 
> In the meantime, you can switch back to using `text-davinci-003` via https://github.com/transitive-bullshit/chatgpt-api/releases/tag/v4.2.0

Thank you for the **blazingly fast** response :)

**Comment by transitive-bullshit** [comment]
> > We're working on a workaround.
> > In the meantime, you can switch back to using `text-davinci-003` via [`v4.2.0` (release)](https://github.com/transitive-bullshit/chatgpt-api/releases/tag/v4.2.0)
> 
> Thank you for the **blazingly fast** response :)

I promise I'm not a bot 😂 hahahha

**Comment by gigantino** [comment]
> I promise I'm not a bot joy hahahha

Well that's what a bot would say...!
On a side note, thank you so much for the work you guys are putting into this.

**Comment by SimonBaars** [comment]
I'm just catching up on the linked thread https://github.com/transitive-bullshit/chatgpt-api/issues/292. Some takeaways for newcomers like me:
- `text-davinci-003` is slightly less optimized for the ChatGPT usecase
- The `text-davinci-003` model does incur API costs (!)

**Comment by transitive-bullshit** [comment]
@qwertyuu this model is unfortunately not available to use from OpenAI's API. We were using a previously leaked model that's fine-tuned for chat and was free, but OpenAI just disabled it.

@SimonBaars 100% agreed with your takeaways. Thanks 🙏

**Comment by dvargas135** [comment]
> @qwertyuu the `text-davinci-002-render` model is unfortunately not available to use from OpenAI's API. We were using a previously leaked model that's fine-tuned for chat and was free, but OpenAI just disabled it.
> 
> @SimonBaars 100% agreed with your takeaways. Thanks 🙏

How do you use that model? Do we need to set up another key? I couldn't find how in the docs...

**Comment by nicksav** [comment]
just exactly when I wanted to use it - its down:) what a luck

**Comment by ikechan8370** [comment]
If we finally couldn't find an available and free chat model, can we turn back to the browser based solution?

**Comment by graid2030** [comment]
I agree. It looks like it would be better to continue the work based on 3.5.2

**Comment by RevAtN** [comment]
> If we finally couldn't find an available and free chat model, can we turn back to the browser based solution?

Agree. ChatGPT Plus is  also free and faster, perhaps even faster than api, may be there is a solution? I can't wait for the official ChatGPT API

**Comment by graid2030** [comment]
OpenAI seems to be trolling us. text-chat-davinci-002-20221122 works again

**Comment by cvpv** [comment]
> OpenAI seems to be trolling us. text-chat-davinci-002-20221122 works again

I just used it. It is still not as good as before

**Comment by basetrade** [comment]
> OpenAI seems to be trolling us. text-chat-davinci-002-20221122 works again

really? Will it have something to do with Microsoft opening up many public chatgpt applications

**Comment by ikechan8370** [comment]
When using ` text-chat-davinci-002-20221122`, it always returned 503 error: That model is currently overloaded with other requests. I remember that the response of this model was relatively fast before. Did OpenAI reduce the computing resources that this model can call?

**Comment by tyanbiao** [comment]
> When using ` text-chat-davinci-002-20221122`, it always returned 503 error: That model is currently overloaded with other requests. I remember that the response of this model was relatively fast before. Did OpenAI reduce the computing resources that this model can call?

same issue

**Comment by xieccc** [comment]
Currently the more stable is the text-davinci-003 model used in version 4.2.0

**Comment by graid2030** [comment]
> Currently the more stable is the text-davinci-003 model used in version 4.2.0

Yes, but the tiny detail is that it is a paid model.

**Comment by nevo-david** [comment]
Is text-davinci-003 the chatgpt model (gpt3)? or is the chatgpt(gpt3.5 instruct)?

**Comment by xieccc** [comment]
> Is text-davinci-003 the chatgpt model (gpt3)? or is the chatgpt(gpt3.5 instruct)?

GPT-3 model. [link](https://platform.openai.com/docs/models/gpt-3)

**Comment by nevo-david** [comment]
So it's not a solution 😛
it's just the old crappy model 🙈

**Comment by nevo-david** [comment]
I am expecting that it's not working when there is high usage on the ChatGPT website.
I was able to generate some a few hours ago.

**Comment by tluyben** [comment]
> We're working on a workaround.
> 
> In the meantime, you can switch back to using `text-davinci-003` via https://github.com/transitive-bullshit/chatgpt-api/releases/tag/v4.2.0

Are you for hire :) You seem to be an absolute top talent at whack-a-mole. Which is something every software engineer needs to be but hardly ever is skilled at.

But yes, same issue. They are also whacking moles on their side. Hope they release the API really soon; I don't mind paying, but davinci-003 is really not as good for what I use it for. And that's putting it mildly.

**Comment by transitive-bullshit** [comment]
> > We're working on a workaround.
> > In the meantime, you can switch back to using `text-davinci-003` via [`v4.2.0` (release)](https://github.com/transitive-bullshit/chatgpt-api/releases/tag/v4.2.0)
> 
> Are you for hire :) You seem to be an absolute top talent at whack-a-mole. Which is something every software engineer needs to be but hardly ever is skilled at.
> 
> But yes, same issue. They are also whacking moles on their side. Hope they release the API really soon; I don't mind paying, but davinci-003 is really not as good for what I use it for. And that's putting it mildly.

You can follow & DM me on [my twitter](https://twitter.com/transitive_bs) w/ a proposal. It has to be organized and very well paid for me to consider it.

**Comment by transitive-bullshit** [comment]
> > I am expecting that it's not working when there is high usage on the ChatGPT website.
> > I was able to generate some a few hours ago.
> 
> It seems to be much worse than it was; as someone else said, it's probably MS opening up the floodgates. It's not easy straightforward to scale yet as your bog standard web app.

My best guess is that they've added ACLs on their end to restrict access to certain models.

**Comment by transitive-bullshit** [comment]
Feel free to join our ChatGPT Hackers Discord if you're currently experiencing issues. https://discord.gg/v9gERj825w

Lots of other hackers in there working on this problem as well.

**Comment by nick4fake** [comment]
@transitive-bullshit I am sorry, but using text-davinci-003 is not a workaround. It has nothing to do with ChatGPT, it's just old stupid gpt3 model, and if anyone can use it as "workaround" then they can just use official api and official library. The whole point is to use revolutionary ChatGPT, not old generative models (which are fine for some usecases, but I don't quite understand what's the point if anyone can just directly use official GPT3 api).

The only real workaround might be using browser-based version, but it looks like you don't plan to support it anymore.

**Comment by transitive-bullshit** [comment]
@nick4fake we provide multiple options and I've pushed 80+ releases over the past two months to try and help people use ChatGPT until the official API is released – at which point we'll replace the underlying impl with the real version.

TL;DR you probably want to use ChatGPTUnofficialProxyAPI which uses the real ChatGPT.

![CleanShot 2023-02-19 at 02 25 54@2x-opt](https://user-images.githubusercontent.com/552829/220491436-c69b9fa7-f215-4618-badb-be27064e897c.jpg)

**Comment by AllanOricil** [comment]
@transitive-bullshit does it work with Open AI Key?  I'm confused because in the README you say "ChatGPT key" but in the code example you wrote `OPENAI_ACCESS_KEY`
![image](https://user-images.githubusercontent.com/55927613/220734811-a1229974-d4c7-4ce2-9066-8d6dbb6d0f54.png)

**Comment by AllanOricil** [comment]
If you like to work with Node, you can now use my implementation of 
this [python lib ](https://github.com/acheong08/OpenAIAuth) to get open ai tokens that work with chatgpt

Github Repository: https://github.com/AllanOricil/chat-gpt-authenticator
npm package: https://www.npmjs.com/package/chat-gpt-authenticator?activeTab=readme

@transitive-bullshit you can add this node version to ur README if you like
